pyrokinetics/local_geometry/LocalGeometryFourier.py

```diff
@@ -0,0 +1,662 @@
+
+            """
+            R_fit, Z_fit, = flux_surface(
+                kappa=kappa,
+                Rcen=R_major,
+                rmin=r_minor,
+                theta=theta,
+                thetaR=thetaR,
+                Zmid=Zmid,
+            )
+            plt.plot(R_fit, Z_fit, label="Fit")
+            plt.plot(R, Z, "--", label="Data")
+            plt.xlabel("R")
+            plt.ylabel("Z")
+            plt.legend()
+            ax = plt.gca()
+            ax.set_aspect("equal")
+            plt.show()
+    
+            # Plot dthetaR_dtheta
+            plt.plot(theta, self.thetaR, label="Fit")
+            plt.plot(theta, thetaR, "--", label="Data")
+            plt.xlabel("theta")
+            plt.ylabel("thetaR")
+            plt.legend()
+            plt.show()
+    
+            # Plot dthetaR_dtheta
+            plt.plot(theta, self.dthetaR_dtheta, label="Fit")
+            plt.plot(theta, np.gradient(thetaR, theta), "--", label="Data")
+            plt.xlabel("theta")
+            plt.ylabel("dthetaR / dtheta")
+            plt.legend()
+            plt.show()
+            """
+            params = [shift, dZ0dr, 1.0, *[0.0] * self.n_moments * 2]
+
+            fits = least_squares(self.minimise_b_poloidal, params)
+
+            # Check that least squares didn't fail
+            if not fits.success:
+                raise Exception(
+                    f"Least squares fitting in Fourier::load_from_eq failed with message : {fits.message}"
+                )
+
+            if verbose:
+                print(f"Fourier :: Fit to Bpoloidal obtained with residual {fits.cost}")
+
+            if fits.cost > 1:
+                import warnings
+
+                warnings.warn(
+                    f"Warning Fit to Fourier in Miller::load_from_eq is poor with residual of {fits.cost}"
+                )
+
+            self.shift = fits.x[0]
+            self.dpsidr = fits.x[1]
+            self.dZ0dr = fits.x[2]
+            self.dasym_dr = fits.x[3 : self.n_moments + 3]
+            self.dsym_dr = fits.x[self.n_moments + 3 :]
+
+            self.dthetaR_dr = self.get_dthetaR_dr(
+                self.theta, self.dasym_dr, self.dsym_dr
+            )
+
+            bpol_fit = get_b_poloidal(
+                kappa=self.kappa,
+                r=self.r_minor,
+                shift=self.shift,
+                dpsidr=self.dpsidr,
+                dZ0dr=self.dZ0dr,
+                R=self.R,
+                Z=self.Z,
+                theta=self.theta,
+                thetaR=self.thetaR,
+                dthetaR_dtheta=self.dthetaR_dtheta,
+                dthetaR_dr=self.dthetaR_dr,
+            )
+
+            plt.plot(self.theta, bpol_fit, label=f"N moments={n_moments}")
+
+        plt.plot(self.theta, self.b_poloidal, "--", label="Data", color="k")
+        plt.legend()
+        plt.xlabel("theta")
+        plt.title("Fit to poloidal field with different number of moments")
+        plt.ylabel("Bpol")
+        plt.show()
```

Comment:
Just here for debugging? It might be nice to provide some separate plotting routines

Reply:
All the plotting is definitely for debugging but I agree some routines might be nice